firefox-ios/Client/TabManagement/Legacy/LegacyTabManager.swift

```diff
@@ -626,7 +632,8 @@ class LegacyTabManager: NSObject, FeatureFlaggable, TabManager, TabEventHandler
 
     func removeTabs(_ tabs: [Tab]) {
         for tab in tabs {
-            self.removeTab(tab, flushToDisk: false)
+            // TODO Is flushToDisk true safe here? Needed for tab refactor disabled (legacy), single tab removal saving state
+            self.removeTab(tab, flushToDisk: true)
```

Comment:
Hmm, this solves the problem but I think it introduces performance issues when we close many tabs with the way the code is functioning currently on the tab tray refactor. It is quite slow if I open up 50+ test tabs. I experimented with dismissing the tab tray earlier but this introduces some funny visual effects while it tries to save so many tabs session data, and with the way the code is now in main you would just wait a long time for everything to close in the tab tray before it's dismissed.

I'm going to think about how to solve this some more this week and see if I can think of a solution to discuss with Orla next week when she's back, seems like the synchronicity issue is kind of the bottleneck here, but I'm not sure.

For reference, this is what it looks like even if I dismiss the tab tray earlier.
https://github.com/mozilla-mobile/firefox-ios/assets/5545720/8013ac7b-78b3-41c9-8af3-7aab8aede571

Reply:
Thanks for pointing that out @dataports ! When I test this case on main I don't see the flickering but the tab tray also hangs for a while without feedback.

Since this specific line only fixes one edge case (single tab closed and restored in the legacy implementation, it could be removed especially if the intention is to eventually roll out the entire tab tray refactor.

Let me know if you could use any help digging into the synchronicity issue. Like Matt's PR, my original solution was looking better before that one PR got rolled back so it seems to have something to do with the change to an asynchronous way of doing things.

One of my alternative ideas for this bug was to start adding completion handlers or awaiting asynchronous session saves, but I felt like I was fighting against whatever the current pattern was and looked for a less invasive solution.

Reply:
@dataports Just was returning to this branch since the synchronous tab selection PR got re-merged.

I actually get the exact behaviour from your sample video on `main` right now, so it does not appear to be something caused by this PR's changes. 🤔  Should we make a performance related ticket for this?

On a separate note, I'll back out of this specific line since we are no longer supporting the legacy tab manager.

Reply:
Made a ticket for the flashing tab close [JIRA](https://mozilla-hub.atlassian.net/browse/FXIOS-9474) and [Github](https://github.com/mozilla-mobile/firefox-ios/issues/20967)

Reply:
Hmm, probably another issue related to `favicons` 😭 generally we have bugs open for them flashing in a bunch of contexts